### PR TITLE
Added epoc to dictionary converter (for human readable display when only given an epoc time)

### PR DESCRIPTION
**Pull request by Razzlegames**
Useful for when user is storing time as epoc and wants to do operations on this
time and then display in human readable form

https://www.facebook.com/groups/godotengine/permalink/737469773056286/?comment_id=738011009668829&reply_comment_id=738192799650650&notif_t=group_comment_reply

### CODE REVIEW

**Comment by Razzlegames** [comment]
Example on usage: 
https://gist.github.com/Razzlegames/a3b5b54188d7994bccb6

**Comment by blurymind** [comment]
Sweey! Thank you thank you thank you! Made my day :)

**Comment by Razzlegames** [comment]
Hold on before merging a few fixes on year calc.  Making another pull.

**Comment by blurymind** [comment]
Thank you for implementing it and also for the python course referal. Much appreciated!

**Comment by Razzlegames** [comment]
No worries friend!  I found the problem.  I was using gmtime as reference for this, and their tm_year	is defined as "years since 1900" lol, so I was getting 116 for year, naturally. Correcting

File: core/bind/core_bind.cpp [hunk 1]
```diff
@@ -490,7 +498,66 @@ Dictionary _OS::get_time(bool utc) const {
...
+	const unsigned int _ytab[2][12] = {
+		{ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 },
+		{ 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 }
+	};
+
+	while (dayno >= _ytab[LEAPYEAR(year)][(size_t)date.month]) {
+		dayno -= _ytab[LEAPYEAR(year)][(size_t)date.month];
+		size_t imonth = static_cast<size_t>(date.month);
+		imonth++;
+		date.month = static_cast<OS::Month>(imonth);
```
Comment (Razzlegames) [inline core/bind/core_bind.cpp]: This is silly.  I'm optimizing this away and just using imonth here.

**Comment by blurymind** [comment]
I hope this gets merged to trunk/main eventualy :)

**Comment by blurymind** [comment]
Thank you for the merge! I added another feature request - this time to convert from a custom user input time to unix time stamp:
https://github.com/godotengine/godot/issues/4038

**Pull request merged**